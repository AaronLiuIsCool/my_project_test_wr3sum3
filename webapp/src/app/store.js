import { configureStore } from '@reduxjs/toolkit';
import resumeReducer from 'features/SmartResume/slicers/resumeSlicer';
import counterReducer from 'features/CounterExample/counterSlice.js';

export default configureStore({
  reducer: {
    counter: counterReducer,
    resume: resumeReducer
  },
});
