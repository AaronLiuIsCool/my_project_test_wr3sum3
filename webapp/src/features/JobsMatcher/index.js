import React, { Suspense } from 'react';
import { Redirect } from 'react-router-dom';

import LoadFallbackComponent from 'components/LoadFallbackComponent';
import './styles/index.scss';

const JobMatcher = React.lazy(() => import('./components/JobsMatcher'));

const LazyLoadJobMatcher = ({ location }) => {
  const query = new URLSearchParams(location.search);
  const resumeId = query.get('resume');
  const page = query.get('page');

  if (!resumeId) {
    return <Redirect to='/' />;
  }

  return (
    <Suspense fallback={<LoadFallbackComponent />}>
      <JobMatcher resume={resumeId} page={page} />
    </Suspense>
  );
};

export default LazyLoadJobMatcher;
