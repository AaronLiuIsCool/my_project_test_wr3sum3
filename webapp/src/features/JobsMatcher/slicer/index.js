import { createSlice } from '@reduxjs/toolkit';

const jobsMatcherSlicers = createSlice({
  name: 'jobs',
  initialState: {
    bookmarked: []
  },
  reducers: {
    addBookmark: (state, action) => {
      const jobUUID = action.payload;
      if (!jobUUID) {
        return;
      }
      state.bookmarked.indexOf(jobUUID) === -1 && state.bookmarked.push(jobUUID);
    },
    removeBookmark: (state, action) => {
      const forRemoval = action.payload;
      if (!forRemoval) {
        return;
      }
      state.bookmarked = state.bookmarked.filter(jobUUID => jobUUID !== forRemoval);
    },
    setBookmarkedList: (state, action) => {
      state.bookmarked = action?.payload || [];
    }
  },
});

export const { addBookmark, removeBookmark, setBookmarkedList } = jobsMatcherSlicers.actions;

export const selectBookmarkedJobs = ({jobs}) => jobs.bookmarked;

export default jobsMatcherSlicers.reducer;
