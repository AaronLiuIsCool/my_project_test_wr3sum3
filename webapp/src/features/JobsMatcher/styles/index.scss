.job-matcher {
  margin: auto;
  width: 1000px;

  // TODO: refactor below
  .form_item {
    label {
      color: #bfc4d3;
    }

    input {
      height: 41px;
      background-color: #f2f4f9;
      color: #333333;
      border-radius: 8px;
      border: none;
      
      ::placeholder, :-ms-input-placeholder, ::-ms-input-placeholder  {
        color: #d5d9e6;
      }
    }
  }

  ul.pagination {
    .page-link {
      width: 40px;
      height: 40px;
      border-radius: 8px;
      border-color: #f2f4f9;
      background-color: #f2f4f9;
      margin-right: 16px;
      color: #bfc4d3;
      text-align: center;
    }

    .active {
      .page-link {
        border-color: #bfc4d3;
        background-color: #bfc4d3;
        color: #ffffff;
      }
    }
  }

  circle {
    transition: stroke-dashoffset 0.35s;
    transform: rotate(-90deg);
    transform-origin: 50% 50%;
  }

  .score-ring {
    margin-top: 4px;
    position: absolute;
  }
}