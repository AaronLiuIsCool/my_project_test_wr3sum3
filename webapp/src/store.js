import { configureStore } from '@reduxjs/toolkit';

import appReducer from 'features/App/slicer';
import accountReducer from 'features/App/slicer/account';
import resumeReducer from 'features/SmartResume/slicer';
import jobsMatcherReducer from 'features/JobsMatcher/slicer';

export default configureStore({
  reducer: {
    app: appReducer,
    account: accountReducer,
    resume: resumeReducer,
    jobs: jobsMatcherReducer,
  },
});
