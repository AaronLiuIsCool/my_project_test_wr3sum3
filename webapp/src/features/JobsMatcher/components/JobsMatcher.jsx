import React, { Fragment, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

import { getLogger } from 'shell/logger';
import ResumeServices from 'shell/services/ResumeServices';
import MatchingServices from 'shell/services/MatchingServices';
import { I8nContext } from 'shell/i18n';
import { selectLanguage, selectUserId } from 'features/App/slicer';

import JobRefinement from './Jobs/JobRefinement';
import SearchHeader from './SearchHeader';
import Jobs from './Jobs';

import zh from '../i18n/zh.json';
import en from '../i18n/en.json';

const logger = getLogger('JobsMatcher');
const matchingServices = new MatchingServices();
const resumeServices = new ResumeServices();

async function getResumeMatchingInfo(resumeId) {
    let resumeDto = {};
    try {
        const response = await resumeServices.getResumeMatchingInfo(resumeId);
        resumeDto = await response.json();
    } catch (exception) {
        logger.error(exception)
    } finally {
        return resumeDto;
    }
}

async function findMatchingJobs(query, country, city, pageNumber) {
    if (!query && !country && !city) {
        return await matchingServices.findMatchingJobs(0);
    }
    return await matchingServices.searchJobs(query, country, city, pageNumber);
}

const JobMatcher = ({ resume }) => {
    const language = useSelector(selectLanguage);
    const messages = language === 'zh' ? zh : en;

    const userId = useSelector(selectUserId);
    const [searchResults, setSearchResults] = useState({});
    const [resultsPageNumber, setResultsPageNumber] = useState(0);
    const [query, setQuery] = useState();
    const [country, setCountry] = useState();
    const [city, setCity] = useState();
    const [ready, setReady] = useState(false);

    const [modalOpened, setModalOpened] = useState(false); 

    const search = async (query, country, city, pageNumber) => {
        try {
            const response = await findMatchingJobs(query, country, city, pageNumber);
            const data = await response.json();
            const jobList = data.success ? data.jobList : {};
            setSearchResults(jobList);
        } catch (exception) {
            logger.error(exception);
        }
    }

    const handleSearch = (query, country, city) => {
        setSearchResults({});
        setResultsPageNumber(0);
        setQuery(query);
        setCountry(country);
        setCity(city);

        search(query, country, city, 0);
    };

    const handlePageChange = (pageNumber) => {
        setResultsPageNumber(pageNumber);
        search(query, country, city, pageNumber);
    }

    useEffect(() => {
        getResumeMatchingInfo(resume).then((resumeDto) => {
            matchingServices.setContext({...resumeDto, userId});
            handleSearch();
            setReady(true);
        });
    }, []); // eslint-disable-line

    if (!ready) {
        return Fragment;
    }

    return (
        <I8nContext.Provider value={messages}>
            <div className="features job-matcher">
                <SearchHeader onSearch={handleSearch} initial={{query, country, city}} />
                <Jobs data={searchResults} pageNumber={resultsPageNumber}
                    onPageChange={handlePageChange} modalOpenHandler={setModalOpened}/>
                {modalOpened && <JobRefinement data={searchResults} modalOpenHandler={setModalOpened} />}
            </div>
        </I8nContext.Provider>
    );
};

JobMatcher.propTypes = {
    resume: PropTypes.string
};

export default JobMatcher;