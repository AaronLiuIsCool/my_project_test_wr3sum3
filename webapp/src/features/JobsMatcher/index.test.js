import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import store from 'store';

import JobsMatcher from './index';


test('renders lazy loaded component for JobsMatcher', () => {
  const jobsMatcher = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <JobsMatcher location={{}} />
      </BrowserRouter>
    </Provider>
  ).toJSON();

  expect(jobsMatcher).toMatchSnapshot();
});
