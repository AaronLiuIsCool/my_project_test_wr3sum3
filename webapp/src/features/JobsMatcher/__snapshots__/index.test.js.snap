// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders lazy loaded component for JobsMatcher 1`] = `null`;
